kubernetes-membership-scheme/src/main/java/org/wso2/carbon/membership/scheme/kubernetes/KubernetesMembershipScheme.java

```diff
@@ -150,13 +155,23 @@ private Parameter getParameter(String name) {
             } catch (InterruptedException ignored) {
             }
             HazelcastUtil.sendMessagesToMember(messageBuffer, member, carbonCluster);
+            if (log.isInfoEnabled()) {
+                log.info(String.format("Current member list: %s",
+                        nwConfig.getJoin().getTcpIpConfig().setEnabled(true).getMembers()));
+            }
         }
 
         @Override public void memberRemoved(MembershipEvent membershipEvent) {
             Member member = membershipEvent.getMember();
             carbonCluster.memberRemoved(member);
             log.info(String.format("Member left: [UUID] %s, [Address] %s", member.getUuid(),
                     member.getSocketAddress().toString()));
+            nwConfig.getJoin().getTcpIpConfig().setEnabled(true)
```

Comment:
WRONG!!! I clearly explained that you need to call the k8s API server, and get the list of pods and see whether that member has lost. This change will cause members to not rejoin after a temporary network glitch.

Reply:
fixed with 7913ee0074c4ebbc8590adf1013ffb6eea5efee6